[WinstonJin] iP

WinstonJin opened the pull request:
# Winde Chatbot

> "One of the Chatbots of _all time_! ~~Just kidding~~" ~An avid user of Winde Chatbot

The features:
- Keep **track** of things you need to do
- Keep **track** of deadlines and events happening on specific dates
- **Remind** you of what you have on which date

So what are you waiting for?
1. **Download** it right now from [here](https://github.com/WinstonJin/ip)
2. Set it up and open it
3. Input your tasks! And just wait for the reminders :sunglasses:

Alternatively, you can input the following line in your ```terminal``` to run it!
```
java -jar Winde.jar
```

If you would like to remove the Winde that appears in ASCII art when you open up the chatbot, you can edit this
greet() method in the Ui class!
``` ruby
    public static void greet() {
        System.out.println("__        __              _");
        System.out.println("\\ \\      / /(_) _ __   __| | ___ ");
        System.out.println(" \\ \\ /\\ / / | || '_  \\/ _` |/ _ \\ ");
        System.out.println("  \\ V  V /  | || | | || (_ || __/ ");
        System.out.println("   \\_/\\_/   |_||_| |_|\\__,_|\\___| ");
        System.out.println("Hello! I'm Winde\n" + "What can I do for you?");
    }
```

Thank you and leave a quick review for us!
- [ ] Easy to Set up
- [x] Easy to Use
- [x] No bugs

Thank you!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

WinstonJin committed:
Add src/main/java/Winde.java

WinstonJin committed:
Add src/main/java/Winde.java

WinstonJin committed:
Add src/main/java/Winde.java

WinstonJin committed:
Add src/main/java/Winde.java

WinstonJin committed:
Add src/main/java/Winde.java

WinstonJin committed:
Add src/main/java/Winde.java

WinstonJin committed:
Add src/main/java/Winde.java

WinstonJin committed:
Add src/main/java/Winde.java

WinstonJin committed:
Add src/main/java/Winde.java

WinstonJin committed:
Add src/main/java/Winde.java

WinstonJin committed:
Add src/main/java/Winde.java

WinstonJin committed:
Add src/main/java/WindeTasks.txt

WinstonJin committed:
Merge branch 'branch-level-7'

WinstonJin committed:
Add src/main/java/Winde.java

WinstonJin committed:
Add scr/main/java/Winde.java

WinstonJin committed:
Add scr/main/java/Winde.java

WinstonJin committed:
Merge branch 'branch-Level-8'

WinstonJin committed:
Add src/main/java/Winde/Winde.java

WinstonJin committed:
Add src/main/java/Winde

WinstonJin committed:
Add src/main/java/Winde

WinstonJin committed:
Add src/main/java/Winde

WinstonJin committed:
Add src/main/java

WinstonJin committed:
Add text-ui-test

WinstonJin committed:
Merge branch 'add-gradle-support'

WinstonJin committed:
Add src/main/java

WinstonJin committed:
Add src/main/java

WinstonJin committed:
Add src/main/java

WinstonJin committed:
Add src/main/java

WinstonJin committed:
Add src/main/java

WinstonJin committed:
Add src/main/java

WinstonJin committed:
Merge branch 'branch-A-JavaDoc'

WinstonJin committed:
Add src/main/java/WindeBot/Parser.java

WinstonJin committed:
Add src/main/java/Commands/DateCommand.java

WinstonJin committed:
Add .

WinstonJin committed:
Update to adhere with coding standards

WinstonJin committed:
Merge branch 'branch-A-CheckStyle'

WinstonJin committed:
Merge to master and tested CheckStyle

WinstonJin committed:
Add GUI Part 4

WinstonJin committed:
Add GUI Part 5

WinstonJin committed:
Merge branch 'branch-Level-10'

brendanng7 commented on src/main/java/windebot/Parser.java:
```diff
@@ -0,0 +1,127 @@
+package windebot;
+
+import commands.AddDeadline;
+import commands.AddEvent;
+import commands.AddMark;
+import commands.AddTodo;
+import commands.AddUnmark;
+import commands.ByeCommand;
+import commands.Command;
+import commands.DateCommand;
+import commands.ErrorCommand;
+import commands.FindCommand;
+import commands.ListCommand;
+import commands.RemoveCommand;
+import exceptions.UnsupportedCommandException;
```
I like that you explicitly imported all the classes

WinstonJin replied:
Haha yea had to from the CheckStyle

brendanng7 commented on src/main/java/windebot/Parser.java:
```diff
@@ -0,0 +1,127 @@
+        String arguments = (parts.length > 1) ? parts[1] : "";
+        Command commands = new ErrorCommand();
+        try {
+            switch (command) {
+            case LIST:
+                commands = new ListCommand();
+                break;
+            case DELETE:
+                commands = new RemoveCommand();
+                break;
+            case MARK:
+                commands = new AddMark();
+                break;
+            case UNMARK:
+                commands = new AddUnmark();
+                break;
+            case TODO:
+                commands = new AddTodo();
+                break;
+            case DEADLINE:
+                commands = new AddDeadline();
+                break;
+            case EVENT:
+                commands = new AddEvent();
+                break;
+            case DATE:
+                commands = new DateCommand();
+                break;
+            case FIND:
+                commands = new FindCommand();
+                break;
+            case BYE:
+                commands = new ByeCommand();
+                break;
+            case UNKNOWN:
+                commands = new ErrorCommand();
+                break;
+            default:
+                throw new UnsupportedCommandException("TYPE /HELP FOR HELP STOOPIDD");
+            }
```
I like that the coding standard is followed when it comes to switch-case.

WinstonJin replied:
I got confused at the start cause i didnt know how to configure my IDE to change it so the case doesnt indent

brendanng7 commented on src/main/java/commands/AddMark.java:
```diff
@@ -0,0 +1,52 @@
+     * Gets the type of command: Add, ChangeMark or Delete
+     */
+
+    public String whatCommand() {
```
Perhaps the naming of this method could be getCommandType.

WinstonJin replied:
Oh yea fair enough

brendanng7 commented on src/main/java/windebot/Winde.java:
```diff
@@ -0,0 +1,123 @@
+        return commandType;
+    }
+
+    public String hello() {
+        String output = "__        __              _" + "\n";
+        output += "\\ \\      / /(_) _ __   __| | ___ " + "\n";
+        output += " \\ \\ /\\ / / | || '_  \\/ _` |/ _ \\ " + "\n";
+        output += "  \\ V  V /  | || | | || (_ || __/ " + "\n";
+        output += "   \\_/\\_/   |_||_| |_|\\__,_|\\___| " + "\n";
+        output += "Hello! I'm Winde\n" + "What can I do for you?" + "\n";
+        return output;
+    }
```
I like the ASCII art!

WinstonJin replied:
Thanks! I spent like 1.5hrs on it haha

brendanng7 replied:
hahaha there are ascii art generators online!

brendanng7 commented on src/main/java/windebot/History.java:
```diff
@@ -0,0 +1,109 @@
+ */
+
+public class History {
+    private static final String WINDE_FILE = "./src/main/java/WindeTasks.txt";
```
I like that you followed the coding standard of constants.

WinstonJin replied:
Thanks!

brendanng7 reviewed (commented):
hmm I'm not sure if we are supposed to push .class files into our repo. But other than that it LGTM.

psh12320 commented on src/main/java/commands/AddUnmark.java:
```diff
@@ -0,0 +1,52 @@
+ * If the input is invalid or the task cannot be found, an exception is thrown.
+ */
+
+public class AddUnmark extends Command {
```
Looks great! Perhaps you can consider changing the class name to reflect that a task is being 'unmarked' instead of using 'Add'.

psh12320 commented on src/main/java/gui/DialogBox.java:
```diff
@@ -0,0 +1,80 @@
+        return new DialogBox(text, img);
+    }
+
+    private void changeDialogStyle(String commandType) {
```
I like the use of different style classes for different commands! 😄

psh12320 commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,88 @@
+ */
+
+public class Task {
+    protected String action;
```
Perhaps consider changing the `action` variable name to `description` to match the JavaDoc header?

psh12320 commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,88 @@
+
+public class Task {
+    protected String action;
+    protected boolean complete;
```
To align with boolean variable naming conventions, could we rename `complete` to `isComplete`? This would help linters verify that this style rule is being followed. 😄

psh12320 commented on src/main/java/windebot/History.java:
```diff
@@ -0,0 +1,109 @@
+            Task task;
+            while (line != null) {
+                if (line.startsWith("T")) {
+                    String[] instr = line.split(" \\| ");
```
I like this bit of code! 👍 Maybe you can consider using a different variable name instead of `instr` and leave it in a plural form for better understanding?

psh12320 commented on src/main/java/windebot/Parser.java:
```diff
@@ -0,0 +1,127 @@
+     * @return The subCommands enum corresponding to the command keyword.
+     */
+
+    private static SubCommandType getCommandType(String input) {
```
I like that you followed the coding style for the _if-else_ class of statements! 😄

psh12320 reviewed (commented):
Overall LGTM. I like that you have been very creative with your bot and using OOP wherever possible!

WinstonJin committed:
Add variable arguements

WinstonJin committed:
Add assert to command

WinstonJin committed:
Add stream to reminder

WinstonJin committed:
Add stream to reminder and test

WinstonJin committed:
Edit code quality to follow code standard

WinstonJin committed:
Merge pull request #1 from WinstonJin/branch-A-Assertions

branch-A-Assertions

WinstonJin committed:
Merge pull request #2 from WinstonJin/master

Master to CodeQuality

WinstonJin committed:
Merge pull request #3 from WinstonJin/master

Master to Streams

WinstonJin committed:
Merge pull request #4 from WinstonJin/branch-A-CodeQuality

branch-A-CodeQuality

WinstonJin committed:
Merge pull request #5 from WinstonJin/master

Master to Streams #2

WinstonJin committed:
Merge pull request #6 from WinstonJin/branch-A-Streams

branch-A-Streams

WinstonJin committed:
Create gradle.yml

WinstonJin committed:
Add update command

WinstonJin committed:
Merge branch 'branch-C-Update'

WinstonJin committed:
Merge branch 'master' of https://github.com/WinstonJin/ip

WinstonJin committed:
Edit ReadMe.md

WinstonJin committed:
Parser: remove argument parameter

Argument Parameter had no use.

Redundant parameter increases code quality

Let's,
* remove it from the parser class

WinstonJin committed:
History: add more javadoc comment

Save method was not clear

Better understanding of save metho

Let's,
* add a better javadoc comment

WinstonJin committed:
Reminder: remove excess code

Code commented out decreases code quality

Code quality increases

Let's,
* remove commented out code

WinstonJin committed:
Ui: remove excess code

Code commented out decreases code quality

Code quality increases

Let's,
* remove commented out code

WinstonJin committed:
Winde: remove excess code

Code commented out decreases code quality

Code quality increases

Let's,
* remove commented out code

WinstonJin committed:
AddTask: Remove assert function

Issue with parsertest

Prevent gradle check from running

WinstonJin committed:
AddEvent: Remove assert function

Issue with parsertest

Prevent gradle check from running

WinstonJin committed:
AddDeadline: Remove assert function

Issue with parsertest

Prevent gradle check from running

WinstonJin committed:
AddTodo: Remove assert function

Issue with parsertest

Prevent gradle check from running

WinstonJin committed:
Winde: Remove Ui

Redundant code reduces code quality

WinstonJin committed:
FindCommand: Change taskNotFound to isTaskFound

Following the Code Quality

WinstonJin committed:
Task: Change complete to isComplete

Following the Code Quality

WinstonJin committed:
Winde: Remove dead code

Following the Code Quality

WinstonJin committed:
History: Reducing code length

Following the Code Quality

WinstonJin committed:
History: Separate the String line and complete

Follow Code Quality

WinstonJin committed:
History: Split the load method to load and readLine

Follow Code Quality

WinstonJin committed:
Parser: Reduce getCommandType method

Follow code quality

WinstonJin committed:
java: Add javac all the java files

WinstonJin committed:
runtest.sh: change directory to correct one

WinstonJin committed:
README: Add custom info of Windebot

WinstonJin committed:
History: Edit the save file directory

WinstonJin committed:
History: Line 24 too long
Followed Code Quality

WinstonJin committed:
runtest.sh: Change directory

WinstonJin committed:
History: Change save method to include correct filepath

WinstonJin committed:
runtest.sh: Add proper filepath

WinstonJin committed:
Gradle: Add proper filepath

WinstonJin committed:
Add more exception handling

WinstonJin committed:
Winde: Change wildcard import to include all exceptions

WinstonJin committed:
More exception handling

WinstonJin committed:
Change runtest.sh

WinstonJin committed:
Change runtest.sh

WinstonJin committed:
Change runtest.sh

WinstonJin committed:
Compatible with Linux and Windows
